lib/ansible/modules/system/gconftool2.py

```diff
@@ -36,146 +38,233 @@
-
-        # If direct is true, create the argument
-        if self.direct:
-            direct = "--direct"
-
-        # Execute the call
-        cmd = "gconftool-2 "
-        try:
-            # If the call is "get", then we don't need as many parameters and
-            # we can ignore some
-            if call_type == 'get':
-                cmd += "--get {0}".format(self.key)
-            # Otherwise, we will use all relevant parameters
-            elif call_type == 'set':
-                cmd += "{0} {1} --type {2} --{3} {4} \"{5}\"".format(direct,
-                                                                     config_source,
-                                                                     self.value_type,
-                                                                     call_type,
-                                                                     self.key,
-                                                                     self.value)
-            elif call_type == 'unset':
-                cmd += "--unset {0}".format(self.key)
-
-            # Start external command
-            process = Popen([cmd], stdout=PIPE, stderr=PIPE, shell=True)
-
-            # In either case, we will capture the output
-            out = process.stdout.read()
-            err = process.stderr.read()
-
-            if len(err) > 0:
-                if fail_onerr:
-                    self.ansible.fail_json(msg='gconftool-2 failed with '
-                                               'error: %s' % (str(err)))
-            else:
-                changed = True
-
-        except OSError:
-            exception = get_exception()
-            self.ansible.fail_json(msg='gconftool-2 failed with exception: '
-                                       '%s' % exception)
-        return changed, out.rstrip()
+from ansible.module_utils.basic import AnsibleModule
+
+
+class GConfTool2(object):
+    """
+    Small wrapper class around the gconftool-2 binary that can be used for
+    manipulating a single preference in GConf repository.
+    """
+
+    def __init__(self, module, direct=False, config_source=None, check_mode=False):
+        """
+        Initialises instance of the class.
+
+        :param module: Ansible module instance used to signal failures and run commands.
+        :type module: AnsibleModule
+
+        :param direct: Specify whether the GConf repository should be accessed directly or through gconfd-2 daemon.
+                       If enabled, valid config_source must be passed in as well.
+        :type direct: bool
+
+        :param config_source: Configuratrion source to use in case direct access has been requested.
+        :type config_source: str
+
+        :param check_mode: Specify whether to only check if a change should be made or if to actually make a change.
+        :type check_mode: bool
+        """
+
+        self.module = module
+        self.base_command = ["gconftool-2"]
```

Comment:
Use [get_bin_path](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L1908) to get path of file